File: bindings/kotlin/whatwg/src/test/kotlin/com/whatwg/WhatWGTests.kt
package com.whatwg

import org.junit.Test
import org.junit.Assert.*

/**
 * Unit tests for WHATWG Platform.
 *
 * Note: Tests that require native library are skipped in unit tests.
 * Integration tests should be run on Android device/emulator.
 */
class WhatWGTests {

    // MARK: - Capability Tests

    @Test
    fun testCapabilityEnumValues() {
        assertEquals(0, Capability.CLIPBOARD.value)
        assertEquals(1, Capability.TIMER.value)
        assertEquals(2, Capability.NETWORK.value)
        assertEquals(3, Capability.STORAGE.value)
    }

    @Test
    fun testAllCapabilities() {
        assertEquals(31, Capability.entries.size)
    }

    // MARK: - JSEngine Tests

    @Test
    fun testJSEngineRecommended() {
        assertEquals(JSEngine.QUICKJS, JSEngine.recommended)
    }

    @Test
    fun testJSEngineDisplayNames() {
        assertEquals("V8", JSEngine.V8.displayName)
        assertEquals("QuickJS", JSEngine.QUICKJS.displayName)
    }

    @Test
    fun testJSEngineValues() {
        assertEquals("v8", JSEngine.V8.value)
        assertEquals("quickjs", JSEngine.QUICKJS.value)
    }

    // MARK: - Exception Tests

    @Test
    fun testWhatWGException() {
        val exception = WhatWGException("Test error")
        assertEquals("Test error", exception.message)
    }

    @Test
    fun testWhatWGExceptionWithCause() {
        val cause = RuntimeException("Cause")
        val exception = WhatWGException("Test error", cause)
        assertEquals("Test error", exception.message)
        assertEquals(cause, exception.cause)
    }
}

File: bindings/swift/Tests/WhatWGTests/WhatWGTests.swift
import XCTest
@testable import WhatWG

final class WhatWGTests: XCTestCase {
    
    // MARK: - Platform Tests
    
    func testPlatformCreation() {
        let platform = WhatWGPlatform()
        XCTAssertNotNil(platform)
    }
    
    func testVersionString() {
        let version = WhatWGPlatform.version
        XCTAssertFalse(version.isEmpty)
    }
    
    func testABIVersion() {
        let version = WhatWGPlatform.abiVersion
        XCTAssertGreaterThan(version, 0)
    }
    
    // MARK: - Capability Tests
    
    func testCapabilityEnumValues() {
        XCTAssertEqual(Capability.clipboard.rawValue, 0)
        XCTAssertEqual(Capability.timer.rawValue, 1)
        XCTAssertEqual(Capability.network.rawValue, 2)
        XCTAssertEqual(Capability.storage.rawValue, 3)
    }
    
    func testAllCapabilities() {
        XCTAssertEqual(Capability.allCases.count, 31)
    }
    
    // MARK: - JSEngine Tests
    
    func testJSEngineRecommended() {
        let recommended = JSEngine.recommended
        #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
        XCTAssertEqual(recommended, .javaScriptCore)
        #else
        XCTAssertEqual(recommended, .quickJS)
        #endif
    }
    
    func testJSEngineDisplayNames() {
        XCTAssertEqual(JSEngine.javaScriptCore.displayName, "JavaScriptCore")
        XCTAssertEqual(JSEngine.v8.displayName, "V8")
        XCTAssertEqual(JSEngine.quickJS.displayName, "QuickJS")
    }
    
    // MARK: - Error Tests
    
    func testErrorDescriptions() {
        XCTAssertNotNil(WhatWGError.initializationFailed.errorDescription)
        XCTAssertNotNil(WhatWGError.contextCreationFailed.errorDescription)
        XCTAssertNotNil(WhatWGError.notInitialized.errorDescription)
        XCTAssertNotNil(WhatWGError.capabilityNotAvailable(.clipboard).errorDescription)
        XCTAssertNotNil(WhatWGError.operationFailed("test").errorDescription)
    }
}
